src/main/scala/io/keen/client/scala/BatchWriterClient.scala

```diff
@@ -0,0 +1,208 @@
+package io.keen.client.scala
+
+import java.util.concurrent.{ Executors, ScheduledThreadPoolExecutor, TimeUnit }
+
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+import com.typesafe.config.{ Config, ConfigFactory }
+
+object BatchWriterClient {
+  val MinSendIntervalEvents: Long = 100
+  val MaxSendIntervalEvents: Long = 10000
+  val MinSendInterval: FiniteDuration = 60.seconds
+  val MaxSendInterval: FiniteDuration = 1.hour
```

Comment:
These bounds seem pretty arbitrary to me—particularly 60-second minimum send interval: in a high-volume scenario a lower interval could still be an important optimization while a minute could be deemed dangerously long.

Putting an upper bound on the queue size by default absolutely makes sense, but that could be achieved with a default value for `queue.send-interval.events` (although its precedence over `send-interval.duration` probably then needs to change somehow). I doubt I'd want anything higher than 10,000, so perhaps that one isn't important, but lower than 100 I could probably imagine in some situations.

Someone could always implement their own batching client to get around these, but is there much value in hardcoding these bound limits versus just having reasonable defaults and some sanity-checking like values >= 0?

On the other hand, you folks know a lot more about Keen IO usage patterns across a broad range of customers than I do…